### PR TITLE
feat: make umap and hdbscan separately configurable

### LINKED ISSUE #666: [gql] HDBScan graphql endpoint to re-compute the clusters

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves #666

This PR is a strawman implementation for the proposed `Pipeline` object.

References to KMeans and t-SNE are only proof-of-concept and are not intended to be functional. 

TODO: caching pipeline results

### CODE REVIEW

File: app/schema.graphql [hunk 1]
```diff
@@ -8,6 +8,19 @@ type Cluster {
   driftRatio: Float
 }
 
+"""Choose one (and only one) of the available options"""
+input ClustersFinder {
```
Comment (mikeldking) [inline app/schema.graphql:12]:
```suggestion
input ClusteringConfig {
```

File: app/schema.graphql [hunk 2]
```diff
@@ -8,6 +8,19 @@ type Cluster {
...
+  """
+  Hierarchical Density-Based Spatial Clustering of Applications with Noise: https://hdbscan.readthedocs.io/en/latest/index.html
+  """
+  hdbscan: HdbscanConfig
+
+  """
+  K-Means Clustering: https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html
+  """
+  kmeans: KmeansConfig
```
Comment (mikeldking) [inline app/schema.graphql:21]: Do we want to bite this off right now?

File: app/schema.graphql [hunk 3]
```diff
@@ -27,6 +40,16 @@ type DataQualityTimeSeries implements TimeSeries {
   data: [TimeSeriesDataPoint!]!
 }
 
+input DataSampler {
+  seed: Int! = 1234567890
+  nSamples: Int! = 1000
+}
```
Comment (mikeldking) [inline app/schema.graphql:46]: nit: the names of these make them feel like classes when they are just parameters. Let's discuss the naming.

Comment (mikeldking) [inline app/schema.graphql:46]:
```suggestion
input EmbeddingDataSamplingConfig {
  seed: Int! = 1234567890
  nSamples: Int! = 1000
}
```

File: app/schema.graphql [hunk 4]
```diff
@@ -144,6 +167,19 @@ type DimensionWithValue {
   value: String
 }
 
+"""Choose one (and only one) of the available options"""
+input DimensionalityReducer {
```
Comment (mikeldking) [inline app/schema.graphql:171]:
```suggestion
"""Choose one (and only one) of the available options"""
input DimensionalityReductionStrategy {
```

File: app/src/pages/embedding/Embedding.tsx [hunk 5]
```diff
@@ -424,7 +422,8 @@ const PointCloudDisplay = ({
                   <PointCloudDisplaySettings />
                 </TabPane>
                 <TabPane name="Hyperparameters">
-                  <PointCloudParameterSettings />
+                  <UMAPParameterSettings />
+                  <HDBSCANParameterSettings />
```
Comment (mikeldking) [inline app/src/pages/embedding/Embedding.tsx:426]: Did you mean to put this here?

File: app/schema.graphql (hunk 2 above)
Comment (RogerHYang) [reply app/schema.graphql:21]: this is just proof of concept, and mostly for testing the data structure, it's not intended to be a working implementation

Comment (mikeldking) [reply app/schema.graphql:21]: nvm I see your comment in the PR. Got me a bit scared there.

File: app/src/pages/embedding/Embedding.tsx (hunk 5 above)
Comment (RogerHYang) [reply app/src/pages/embedding/Embedding.tsx:426]: please feel free to reorganize. i only made it a separate component for my own testing purposes, and will defer to you regarding UI placements

**Comment by RogerHYang** [comment]
close this PR in favor of https://github.com/Arize-ai/phoenix/pull/761

**Pull request closed without merging**